libcontainer/intelrdt/mbm.go

```diff
@@ -0,0 +1,108 @@
+// +build linux
+
+package intelrdt
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+var (
+	// The flag to indicate if Intel RDT/MBM is enabled
+	isMbmEnabled bool
+
+	enabledMonFeatures monFeatures
+)
+
+type monFeatures struct {
+	mbmTotalBytes bool
+	mbmLocalBytes bool
+	llcOccupancy  bool
+}
+
+// Check if Intel RDT/MBM is enabled
+func IsMbmEnabled() bool {
+	return isMbmEnabled
+}
+
+func getMonFeatures(intelRdtRoot string) (monFeatures, error) {
+	file, err := os.Open(filepath.Join(intelRdtRoot, "info", "L3_MON", "mon_features"))
+	if err != nil {
+		return monFeatures{}, err
+	}
+	return parseMonFeatures(file)
+}
+
+func parseMonFeatures(file io.Reader) (monFeatures, error) {
+	s := bufio.NewScanner(file)
+
+	monFeatures := monFeatures{}
+
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return monFeatures, err
+		}
+
+		switch feature := s.Text(); feature {
+
+		case "mbm_total_bytes":
+			monFeatures.mbmTotalBytes = true
+		case "mbm_local_bytes":
+			monFeatures.mbmLocalBytes = true
+		case "llc_occupancy":
+			monFeatures.llcOccupancy = true
+		}
```

Comment:
I would like to add here a default case to catch unsupported/unexpected feature. What do you think?